crates/bevy_app/src/plugin.rs

```diff
@@ -56,19 +89,46 @@ use std::any::Any;
 /// # fn damp_flickering() {}
 /// ````
 pub trait Plugin: Downcast + Any + Send + Sync {
-    /// Configures the [`App`] to which this plugin is added.
-    fn build(&self, app: &mut App);
+    /// Returns required sub apps before finalizing this plugin.
```

Comment:
More docs here about why this is useful to users please.

Reply:
If we keep this, a doc test demonstrating how to require the rendering subapp would be really valuable. This will be by far the most common pattern.